src/core/gui/XournalView.h

```diff
@@ -183,6 +185,13 @@ class XournalView: public DocumentListener, public ZoomListener {
      */
     RepaintHandler* repaintHandler = nullptr;
 
+    /**
+     * Geometry tool, if active
+     */
+    GeometryTool* geometryTool = nullptr;
+    GeometryToolController* geometryToolController = nullptr;
+    GeometryToolHandler* geometryToolHandler = nullptr;
```

Comment:
Maybe moved to other classes (see comment above)

```suggestion
    std::unique_ptr<GeometryTool> geometryTool;
    std::unique_ptr<GeometryToolController> geometryToolController;
    std::unique_ptr<GeometryToolHandler> geometryToolHandler;
```

Reply:
I run into trouble with `unique_ptr`s (related to derived classes and casting), but I can try again.

Reply:
Could you describe the problem you encountered? We might be able to help.

Reply:
Strange. This time it worked at once. Can you please check?